src/java.base/share/classes/java/util/UUID.java

```diff
@@ -462,31 +461,106 @@ public long node() {
      */
     @Override
     public String toString() {
-        int i0 = (int) (mostSigBits >> 32);
-        int i1 = (int) mostSigBits;
-        int i2 = (int) (leastSigBits >> 32);
-        int i3 = (int) leastSigBits;
-
         byte[] buf = new byte[36];
-        HexDigits.put4(buf, 0, i0 >> 16);
-        HexDigits.put4(buf, 4, i0);
         buf[8] = '-';
-        HexDigits.put4(buf, 9, i1 >> 16);
         buf[13] = '-';
-        HexDigits.put4(buf, 14, i1);
         buf[18] = '-';
-        HexDigits.put4(buf, 19, i2 >> 16);
         buf[23] = '-';
-        HexDigits.put4(buf, 24, i2);
-        HexDigits.put4(buf, 28, i3 >> 16);
-        HexDigits.put4(buf, 32, i3);
+
```

Comment:
I would suggest a comment here along the lines of "Although the UUID byte ordering is defined to be big-endian, LittleEndian is used here to optimize for the most common architectures. `hex8` reverses the order internally."